workers/common/src/main/kotlin/common/env/definition/EnvironmentVariableDefinition.kt

```diff
@@ -21,16 +21,31 @@ package org.eclipse.apoapsis.ortserver.workers.common.env.definition
 
 import org.eclipse.apoapsis.ortserver.model.Secret
 
+/**
+ * An interface representing an environment variable definition.
+ */
+sealed interface EnvironmentVariableDefinition {
+    val name: String
+}
+
 /**
  * A class defining an environment variable which needs to be present when analyzing a repository.
  *
  * This is typically used to provide credentials or other values to external tools called during the analysis. The
  * variable can have an arbitrary name. Its value is obtained from a [Secret].
  */
-data class EnvironmentVariableDefinition(
+data class SecretVariableDefinition(
```

Comment:
The documentation is still not ideal. Parts of the documentation from `SecretVariableDefinition` should be moved to the new interface, especially regarding the purpose. The concrete implementation classes should then have specific documentation how they obtain their value. `SimpleVariableDefinition` is still missing documentation.